grug/app/src/app.rs

```diff
@@ -156,6 +156,31 @@ where
             });
         }
 
+        // Remove orphaned codes (those that are not used by any contract) that
+        // have been orphaned longer than the maximum age.
+        if let Some(since) = block
+            .timestamp
+            .into_nanos()
+            .checked_sub(cfg.max_orphan_age.into_nanos())
```

Comment:
It'd be easier if we implement `Math` on `Duration`, which is just a wrapper over `Uint128`.

Reply:
Yeah, doable